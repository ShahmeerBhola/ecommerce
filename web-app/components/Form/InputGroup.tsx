import React, { FC } from 'react';

import ErrorMessage from './ErrorMessage';

type Props = {
    touched?: boolean;
    className?: string;
    error?: Forms.FormikError;
    noBottomPadding?: boolean;
    noXPadding?: boolean;
};

const InputGroup: FC<Props> = ({
    error,
    touched,
    className,
    noBottomPadding = false,
    noXPadding = false,
    children,
}) => {
    const isError = error && touched;
    const classNames = `pt-2 box-border ${noXPadding ? '' : 'px-2'} ${className}`;

    return (
        <div className={`${classNames} ${noBottomPadding ? '' : 'pb-3'}`}>
            <div className="relative h-full">
                {children}
                {isError && <ErrorMessage error={error} />}
            </div>
        </div>
    );
};

export default InputGroup;
