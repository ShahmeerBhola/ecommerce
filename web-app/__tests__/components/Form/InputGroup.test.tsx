import React from 'react';
import renderer from 'react-test-renderer';
import InputGroup from '../../../components/Form/InputGroup';

describe('InputGroup component', () => {
  test('it renders properly with default props', () => {
    const tree = renderer
      .create(
        <InputGroup>
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with className prop', () => {
    const tree = renderer
      .create(
        <InputGroup className="my-default-class">
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it doesnt render error message if touched is false and error is null', () => {
    const tree = renderer
      .create(
        <InputGroup touched={false} error={null}>
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it doesnt render error message if touched is false and error is non-null', () => {
    const tree = renderer
      .create(
        <InputGroup touched={false} error="There is an error!">
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it doesnt render error message if touched is true and error is null', () => {
    const tree = renderer
      .create(
        <InputGroup touched={true} error={null}>
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders error message if touched is true and error is non-null', () => {
    const tree = renderer
      .create(
        <InputGroup touched={true} error="There is an error!">
          <input type="text" />
        </InputGroup>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
