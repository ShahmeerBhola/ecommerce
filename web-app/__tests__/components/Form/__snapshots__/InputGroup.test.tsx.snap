// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component it doesnt render error message if touched is false and error is non-null 1`] = `
<div
  className="pt-2 px-2 box-border pb-5"
>
  <input
    type="text"
  />
</div>
`;

exports[`InputGroup component it doesnt render error message if touched is false and error is null 1`] = `
<div
  className="pt-2 px-2 box-border pb-5"
>
  <input
    type="text"
  />
</div>
`;

exports[`InputGroup component it doesnt render error message if touched is true and error is null 1`] = `
<div
  className="pt-2 px-2 box-border pb-5"
>
  <input
    type="text"
  />
</div>
`;

exports[`InputGroup component it renders error message if touched is true and error is non-null 1`] = `
<div
  className="pt-2 px-2 box-border"
>
  <input
    type="text"
  />
  <p
    className="text-red-100 text-xs pt-1"
  >
    There is an error!
  </p>
</div>
`;

exports[`InputGroup component it renders properly with className prop 1`] = `
<div
  className="pt-2 px-2 box-border pb-5 my-default-class"
>
  <input
    type="text"
  />
</div>
`;

exports[`InputGroup component it renders properly with default props 1`] = `
<div
  className="pt-2 px-2 box-border pb-5"
>
  <input
    type="text"
  />
</div>
`;
